lxqtnotification.cpp

```diff
@@ -173,9 +182,43 @@ void NotificationPrivate::setActions(QStringList actions, int defaultAction)
 
 const Notification::ServerInfo NotificationPrivate::serverInfo()
 {
-    Notification::ServerInfo info;
-    info.name = mInterface->GetServerInformation(info.vendor, info.version, info.specVersion);
-    return info;
+    if (!sIsServerInfoQuried) {
+        queryServerInfo (/*async=*/false);
+    }
+
+    return sServerInfo;
+}
+
+void NotificationPrivate::queryServerInfo(bool async)
+{
+    QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(mInterface->GetServerInformation(), this);
+
+    connect(watcher, &QDBusPendingCallWatcher::finished, [this](QDBusPendingCallWatcher* callWatcher) {
+            Q_Q(Notification);
+            QDBusPendingReply<QString, QString, QString, QString> reply = *callWatcher;
+
+            if (!reply.isError()) {
+                sServerInfo.name         = reply.argumentAt<0>();
+                sServerInfo.vendor       = reply.argumentAt<1>();
+                sServerInfo.version      = reply.argumentAt<2>();
+                sServerInfo.specVersion  = reply.argumentAt<3>();
+            } else {
+                 // server either responds something strange or nothing; assume it's malfunctioning
+                sServerInfo.name.clear();
+                sServerInfo.vendor.clear();
+                sServerInfo.version.clear();
+                sServerInfo.specVersion.clear();
+            }
+            sIsServerInfoQuried = true;
+            emit q->serverInfoReady();
```

Comment:
@Fat-Zer 
Please change `emit` to `Q_EMIT` for it to be compilable. I want to merge your [other PR](https://github.com/lxqt/lxqt-notificationd/pull/142) that's based on this one before the next release.

Reply:
Done and rebased upon master.